### DISCUSSION TITLE
Localization of Simplified Chinese is different between released version and local built version

### DISCUSSION THREAD

**Discussion by bianyun1981** [dbeaver/dbeaver#14914] [DBeaver Development] [answered]
#### System information: 
- Operating system (distribution) and version: Windows 10 Enterprise Edition (Simplified Chinese), 21H1
- DBeaver version: 21.3.1.202112130652
- Additional extensions

#### Describe the problem you're observing:
I am a DBeaver fan, and I want to contribute to the localization of DBeaver (Simplified Chinese). So I decided to fork the DBeaver repo, and clone it and build it yesterday, but when I execute the local built executable file (path:  dbeaver\product\community\target\products\org.jkiss.dbeaver.core.product\win32\win32\x86_64\dbeaver\dbeaver.exe), I found that the localization part is different from the release version (21.3.1.202112130652). I make some comparison pictures for illustration, left side is the released version, right side is the local built version.

![comparison_00](https://user-images.githubusercontent.com/17854699/146138735-37a62818-e0a4-4f7c-8120-eae63aa191a8.jpg)

![comparison_01](https://user-images.githubusercontent.com/17854699/146138759-c6984305-7ae8-414f-a96f-138d5deef066.jpg)

![comparison_02](https://user-images.githubusercontent.com/17854699/146138775-c634ae80-cf39-437c-8bac-a2abca883b36.jpg)

![comparison_03](https://user-images.githubusercontent.com/17854699/146138794-c9ac3ed2-4f2e-4aad-99ea-d2575dcf2108.jpg)

![comparison_04](https://user-images.githubusercontent.com/17854699/146138843-80d2e8f0-bb26-4512-9481-03c49c9d4b6d.jpg)

![comparison_05](https://user-images.githubusercontent.com/17854699/146138870-d3aa9bb4-5fad-46a7-bba1-02c217eec874.jpg)

![comparison_06](https://user-images.githubusercontent.com/17854699/146138891-e9afae68-bc33-4e08-bf91-edcbfe5c1ed1.jpg)

![comparison_07](https://user-images.githubusercontent.com/17854699/146138903-6b67d285-88f6-45da-9b67-37b1ccffe921.jpg)

![comparison_08](https://user-images.githubusercontent.com/17854699/146138939-bef62c17-294e-4314-a469-3f36b6276bcf.jpg)

I try to search some original key text (which has error in localization) from some bundle.properties (listed in a table from https://github.com/dbeaver/dbeaver/wiki/Localization), but none of them can be found, such as (menu 'Edit' -> 'Find Next', menu 'Edit' -> 'Find Previous').

To sum up, I need some help to find out why the local built version is not same as the released version, and why some original key text of l10n  cannot be found in any bundle.properties. Or in short, what is the right steps to contribute to the localization of DBeaver.

**Comment by ShadelessFox** [comment, marked as answer]
Hello @bianyun1981,

The locally built version lacks some localization because it's a part of Eclipse localization.

We use Babel while packaging to include that localization too. You can install it manually.

I'm going to convert this issue to the discussion.

**Reply by bianyun1981** [reply]
Hi, @ShadelessFox,
Thank you for your reply. 

I tried install babel language packs manually to locally built version, and it does works. But I also wondered if there is a way to install babel language packs once and for all, I'd like to see the localization works even when debuging in eclipse, not only after the building finished and then installing babel language packs manually.

**Reply by VolareYang** [reply]
Hello, I also encountered the problem of localization packaging, how did the landlord solve it

**Reply by VolareYang** [reply]
> Hello @bianyun1981,
> 
> The locally built version lacks some localization because it's a part of Eclipse localization.
> 
> We use Babel while packaging to include that localization too. You can install it manually.
> 
> I'm going to convert this issue to the discussion.


Hello，How to automatically integrate Babel packages when packing.

**Reply by VolareYang** [reply]
> Hello,@ShadelessFox,
How to automatically integrate Babel packages when packing.

**Reply by bianyun** [reply]
@VolareYang 

Hi, I think this babel package is integrated into DBeaver release as a internal plugin (maintained by eclipse community) when DBeaver is published, and the release publisher is in charge of setting up the stable version of babel packages plugin.

I think what we can do is to do the locallization work for the DBeaver-specific (that is to say, you should only work on all the localizaiton properties file in the DBeaver source code).  

And if some piece of unlocalized text has no corresponding key defined in properties file, then you should find out whether eclipse babel is reponsible for localization for this piece of text. If this piece of text is releated with DBeaver specific feature, then I think it should be localized in DBeaver propreties files, and these properties file should be created first (usually by committing a new issue and waiting for other contributers to create those properties files, like this https://github.com/dbeaver/dbeaver/issues/21384, and then you can do your localization work).

Hope this helps you.  :)

P.S. @bianyun1981 is my old account, it is deprecated.

**Reply by VolareYang** [reply]
Thank you! @bianyun 
 How do I manually install the babel language pack to a locally built version.
 ask for your help.
